hazelcast-client/src/test/java/com/hazelcast/client/starter/HazelcastClientStarterTest.java

```diff
@@ -0,0 +1,79 @@
+package com.hazelcast.client.starter;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.core.IMap;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.NightlyTest;
+import com.hazelcast.test.starter.HazelcastStarter;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(NightlyTest.class)
+public class HazelcastClientStarterTest {
```

Comment:
Are these tests directly related to this PR?

Reply:
These tests were under hazelcast instead of hazelcast-client. Because the support of client starter with version was under hazelcast. Since client classes are not available there there was usage of reflection and class loading. And the support of client config was missing for clients. 
I cleaned the unnecessary reflection usage and added client config support. While doing that,  I  had to move client related classes and tests to hazelcast-client package.